src/raftstore/store/transport.rs

```diff
@@ -11,13 +11,98 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::raftstore::store::{CasualMessage, PeerMsg, RaftCommand, RaftRouter, StoreMsg};
+use crate::raftstore::{DiscardReason, Error, Result};
+use crossbeam::TrySendError;
 use kvproto::raft_serverpb::RaftMessage;
-
-use crate::raftstore::Result;
+use std::sync::mpsc;
 
 /// Transports messages between different Raft peers.
 pub trait Transport: Send + Clone {
-    fn send(&self, msg: RaftMessage) -> Result<()>;
+    fn send(&mut self, msg: RaftMessage) -> Result<()>;
 
     fn flush(&mut self);
 }
+
+/// Routes message to target region.
+///
+/// This trait should only be used for internal commands. And messages
```

Comment:
It feels ambiguous, what is the `inernal`? Can it be `pub (super)` or other visibility?